Makefile

```diff
@@ -1,5 +1,7 @@
 VERSION = $(shell cat emscripten-version.txt | sed s/\"//g)
-DESTDIR ?= ../emscripten-$(VERSION)
+ifeq ($(DESTDIR),)
+	DESTDIR := $(shell mktemp -d)/emscripten-$(VERSION)
```

Comment:
Isn't the `?=` syntax good enough here?  Do we want to support DESTDIR being empty?

Reply:
Actually, both can handle empty DESTDIR...
The problem is that `?=` will have different values when DESTDIR is needed, whereas `:=` is defined once. This is problematic for `mktemp -d`.
I can't find a one line equivalent of `?=` that handles both empty and defined once...